### PR TITLE
Add GlobalAveragePooling CNN block

### PR DESCRIPTION
**Pull request by blazekadam**

### CODE REVIEW

File: cxflow_tensorflow/models/conv_blocks.py [hunk 1]
```diff
@@ -215,6 +215,28 @@ def __init__(self, ap_fn: Callable=tf.identity, **kwargs):
         super().__init__(prefix='a', pool_fn=ap_fn, **kwargs)
 
 
+class GlobalAveragePoolBlock(BaseBlock):
+    """Global average pooling block for pooling blocks."""
```
Comment (FloopCZ) [inline cxflow_tensorflow/models/conv_blocks.py]: That is a funky docstring. :-)

File: cxflow_tensorflow/models/conv_blocks.py [hunk 2]
```diff
@@ -215,6 +215,28 @@ def __init__(self, ap_fn: Callable=tf.identity, **kwargs):
...
+
+    def __init__(self, extra_dim: Tuple[int]=(), **kwargs):
+        """
+        Try to parse and create new :py:class:`GlobalAveragePoolBlock`.
+
+        :param extra_dim: extra dimension (for 5-dim tensors)
```
Comment (FloopCZ) [inline cxflow_tensorflow/models/conv_blocks.py]: This parameter has a strange semantics. You don't even use the values of the tuple, but only its `len`. Furthermore, it is not really clear (well, at least to me :-)) how it should be used.

File: cxflow_tensorflow/models/conv_blocks.py (hunk 1 above)
Comment (blazekadam) [reply cxflow_tensorflow/models/conv_blocks.py]: Funky indeed. Updated.

File: cxflow_tensorflow/models/conv_blocks.py (hunk 2 above)
Comment (blazekadam) [reply cxflow_tensorflow/models/conv_blocks.py]: Good point. Removed.

File: cxflow_tensorflow/models/conv_blocks.py [hunk 3]
```diff
@@ -215,6 +215,30 @@ def __init__(self, ap_fn: Callable=tf.identity, **kwargs):
         super().__init__(prefix='a', pool_fn=ap_fn, **kwargs)
 
 
+class GlobalAveragePoolBlock(BaseBlock):
+    """
+    Global average pooling block effectively flattening spatial dimensions of the input feature maps.
+
+    .. warning::
+        Expects ?HWC data format (e.g. BHWC or BTHWC).
```
Comment (FloopCZ) [inline cxflow_tensorflow/models/conv_blocks.py]: I know that it may seem absolutely crystal clear to you, but still, I would prefer something like this:
```
Expects either 4D or 5D input tensor (i.e., BHWC or BTHWC)
and returns 3D or 4D tensor (i.e., B1C or BT1C) respectively.
```
and not in a warning window. But if you don't like it, feel free to keep the current version.

Comment (blazekadam) [reply cxflow_tensorflow/models/conv_blocks.py]: Okay, I can see your point nonetheless:

- it is not limited to 4 or 5 dim tensors
- `BHWC` and similar dimension codes are sort of well known among the community (and `tf.layers`, `t.keras` and others use these abbreviations in docs, argument values etc.)

Since other blocks are not exactly well documented and the blocks API is not quite mature, I ll leave it like this...

Comment (blazekadam) [reply cxflow_tensorflow/models/conv_blocks.py]: Thanks for the comment though.

Comment (FloopCZ) [reply cxflow_tensorflow/models/conv_blocks.py]: Sure, your call.

Strange though, I have never seen `BHWC` in any of the `tf` documentation pages. We are probably reading a different docs then. :-)

(In the docs I read, they always tidily use tuple notation with full words and never call it data format.)
